### ISSUE TITLE
Disabling emails causes 500 error when adding new project members

### ISSUE THREAD

**Issue by marlenekoh** [healthchecks/healthchecks#920]
Healthchecks should not raise 500 error when email configuration is not provided. It should not send any emails and fail gracefully.

```
Internal Server Error: /projects/142d4567-6d58-4484-ac27-31b447996fbc/settings/
Traceback (most recent call last):
  File "/lsiopy/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/lsiopy/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/lsiopy/lib/python3.11/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/healthchecks/hc/accounts/views.py", line 428, in project
    if project.invite(user, role=invite_form.cleaned_data["role"]):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/healthchecks/hc/accounts/models.py", line 415, in invite
    profile.send_instant_login_link(membership=m, redirect_url=checks_url)
  File "/app/healthchecks/hc/accounts/models.py", line 146, in send_instant_login_link
    emails.login(self.user.email, ctx)
  File "/app/healthchecks/hc/lib/emails.py", line 84, in login
    send(make_message("login", to, ctx))
  File "/app/healthchecks/hc/lib/emails.py", line 67, in send
    assert settings.EMAIL_HOST, (
AssertionError: No SMTP configuration, see https://github.com/healthchecks/healthchecks#sending-emails
```

happens on any version >=v2.10

**Comment by cuu508** [comment]
Healthchecks is written with the assumption that working SMTP credentials will be available.

Can you share details about your planned use-case? 

* Are you setting up an instance for a homelab or for a company? 
* What's the deciding factor why you don't want to use email?
* Email is used to deliver login links. How do you plan to log users in, using external authentication and  [REMOTE_USER_HEADER](https://healthchecks.io/docs/self_hosted_configuration/#REMOTE_USER_HEADER)?
* Email is used to deliver monthly and weekly reports, and daily or hourly reminders when any checks are down. Are you not planning to use those?
* Email is used to deliver verification code before sensitive actions, would you expect the verification step to be skipped if there is no outgoing email?

**Comment by marlenekoh** [comment]
Thanks for the quick response! The main reason why we don't want to use email is because not all users have emails in our active directory.

Healthchecks is deployed behind kong api gateway. We have configured the [kong request transformer](https://docs.konghq.com/hub/kong-inc/request-transformer/) to add the `REMOTE_USER_HEADER` depending on the user authenticating to kong (we have also configured kong oidc plugin for authentication). 

Users are encouraged to configure mattermost notifications instead.
